src/client.rs

```diff
@@ -419,6 +585,30 @@ impl Client {
         }
         Ok(())
     }
+
+    pub async fn server_info(&mut self) -> Result<&Value, ClientError> {
```

Comment:
Couldn't we «reify» this and return `self.server_info` if not None?

Reply:
Ah oops. That was the intent, thanks.